actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -268,7 +270,7 @@ def ==(other)
       if other.respond_to?(:permitted?)
         permitted? == other.permitted? && parameters == other.parameters
       else
-        if Hash === other
+        if self.class.allow_deprecated_parameters_hash_equality && Hash === other
           ActiveSupport::Deprecation.warn <<-WARNING.squish
             Comparing equality between `ActionController::Parameters` and a
             `Hash` is deprecated and will be removed in Rails 7.1. Please only do
```

Comment:
We should talk about the config in this message.

Reply:
Thanks, added the following to the warning

```
To disable the deprecated behaviour set
`Rails.application.config.action_controller.allow_deprecated_parameters_hash_equality = false`.
```